src/main/java/org/spongepowered/api/util/reporting/PrettyPrinter.java

```diff
@@ -0,0 +1,468 @@
+
+    /**
+     * Print a formatted representation of the specified throwable with the
+     * default indent (4)
+     *
+     * @param th Throwable to print
+     * @return fluent interface
+     */
+    PrettyPrinter add(Throwable th);
+
+    /**
+     * Print a formatted representation of the specified throwable with the
+     * specified indent
+     *
+     * @param th Throwable to print
+     * @param indent Indent size for stacktrace lines
+     * @return fluent interface
+     */
+    PrettyPrinter add(Throwable th, int indent);
+
+    /**
+     * Print a formatted representation of the specified stack trace with the
+     * specified indent
+     *
+     * @param stackTrace stack trace to print
+     * @param indent Indent size for stacktrace lines
+     * @return fluent interface
+     */
+    PrettyPrinter add(StackTraceElement[] stackTrace, int indent);
+
+    /**
+     * Adds the specified object to the output
+     *
+     * @param object object to add
+     * @return fluent interface
+     */
+    PrettyPrinter add(Object object);
+
+    /**
+     * Adds the specified object to the output
+     *
+     * @param object object to add
+     * @param indent indent amount
+     * @return fluent interface
+     */
+    PrettyPrinter add(Object object, int indent);
+
+    /**
+     * Adds a formatted line to the output, and attempts to wrap the line
+     * content to the current wrap width
+     *
+     * @param format format string
+     * @param args arguments
+     *
+     * @return fluent interface
+     */
+    PrettyPrinter addWrapped(String format, Object... args);
+
+    /**
+     * Adds a formatted line to the output, and attempts to wrap the line
+     * content to the specified width
+     *
+     * @param width wrap width to use for this content
+     * @param format format string
+     * @param args arguments
+     *
+     * @return fluent interface
+     */
+    PrettyPrinter addWrapped(int width, String format, Object... args);
+
+
+    /**
+     * Add a formatted key/value pair to the output
+     *
+     * @param key Key
+     * @param format Value format
+     * @param args Value args
+     * @return fluent interface
+     */
+    PrettyPrinter kv(String key, String format, Object... args);
+
+    /**
+     * Add a key/value pair to the output
+     *
+     * @param key Key
+     * @param value Value
+     * @return fluent interface
+     */
+    PrettyPrinter kv(String key, Object value);
+
+    /**
+     * Set the minimum key display width
+     *
+     * @param width width to set
+     * @return fluent
+     */
+    PrettyPrinter kvWidth(int width);
+
+    /**
+     * Add all values of the specified map to this printer as key/value pairs
+     *
+     * @param map Map with entries to add
+     * @return fluent
+     */
+    PrettyPrinter add(Map<?, ?> map);
+
+    /**
+     * Adds a horizontal rule to the output
+     *
+     * @return fluent interface
+     */
+    PrettyPrinter hr();
+
+
```

Comment:
Don't care about formatting while the entire structure is still in the air. I'm only going to ask for formatting once the PR is nearing closer to being pulled.